src/source-control/GitSourceControl.ts

```diff
@@ -1,9 +1,31 @@
-import type { Commit, CommitHash } from "../shared/types";
+import { func } from "prop-types";
+import type { BranchName, Commit, CommitHash } from "../shared/types";
 import type {
   SourceControl,
   SourceControlRepositoryUpdateListener,
 } from "./SourceControl";
 
+var randomWords = require("random-words");
+let localRefPrefix: string = "refs/heads/";
+
+function refIsLocal(refString: string): boolean {
+  return refString.startsWith(localRefPrefix);
+}
+
+function localRefToBranchName(refString: string): string {
+  if (!refIsLocal(refString)) return refString;
+  return refString.substr(localRefPrefix.length);
+}
+
+function isSttackBranch(branch: BranchName): boolean {
+  if (branch.startsWith("sttack-")) return true;
+  return false;
```

Comment:
nit: will be cleaner to just return the condition here

```suggestion
  return branch.startsWith("sttack-");
```